File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.js
function solution(arr, queries) {
    var answer = [];
    for (let i = 0; i < queries.length; i++) {
        let s = queries[i][0];
        let e = queries[i][1];
        let k = queries[i][2];
        let min = Infinity;
        for (let j = s; j <= e; j++) {
            if (arr[j] > k && arr[j] < min) {
                min = arr[j];
            }
        }
        answer.push(min === Infinity ? -1 : min);
    }
            
    return answer;
}

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.py
def solution(arr, queries):
    answer = []
    for query in queries:
        s, e, k = query
        current_min = float('inf') 
        for j in range(s, e + 1):
            if arr[j] > k and arr[j] < current_min:
                current_min = arr[j]
        if current_min == float('inf'):
            answer.append(-1)
        else:
            answer.append(current_min)
    return answer